src/feature.js

```diff
@@ -151,8 +147,17 @@ class Feature {
     );
   }
 
-  recordSurveyInteraction(tabInfo) {
+  recordSurveyInteraction(tabId, payloadValue, disableStudyChecked) {
+    const tabInfo = TabRecords.getOrInsertTabInfo(tabId);
+    tabInfo.telemetryPayload.page_reloaded_survey = payloadValue;
     browser.storage.local.set({[tabInfo.telemetryPayload.etld]: true});
+    if (disableStudyChecked) {
+      // FIXME ?: race condition:
+      // endStudy may execute before sendTelemetry finishes
+      // can this.sendTelemetry await for browser.study.sendTelemetry ?
+      this.sendTelemetry(tabInfo.telemetryPayload);
```

Comment:
I'm getting an error from this, telemetry is sending in time for me, but since the study is removed tabInfo disappears by the time it gets to `feature.js 139`
We could await, or add an optional callback to sendTelemetry?

Reply:
Hmm ... I think that was actually a bug in the calling `this.recordSurveyInteraction` code was still passing `tabInfo` instead of `tabId`. The refactor needed it to pass `tabId`. See if that fixed it?

If so, no need to worry about the race condition.